CRM/Extendedreport/Form/Report/Grant/Detail.mgd.php

```diff
@@ -1,5 +1,6 @@
 <?php
-if (!in_array('CiviGrant', Civi::settings()->get('enable_components'), TRUE)) {
+$extensionCheck = civicrm_api3('Extension', 'get', ['key' => 'civigrant', 'sequential' => 1])['values'];
+if (!in_array('CiviGrant', Civi::settings()->get('enable_components'), TRUE) || empty($extensionCheck) || $extensionCheck[0]['status'] !== 'installed') {
```

Comment:
Isn't this `if` statement always true? In >5.47 the first clause is true, so true. In <5.47 the second clause is true, so true.

Also the error in 5.47+ occurs because of line 20 down below - there is no component CiviGrant so it tries to insert 0 which violates the foreign key.

Reply:
I thought I stuffed it up some now

Reply:
ug - serves me right for blind-merging....

Reply:
The report will never get created because CiviGrant as the component is not added via the CiviGrant extension